packages/coinjoin/src/utils/http.ts

```diff
@@ -62,25 +67,57 @@ export const coordinatorRequest = async <R = void>(
     const baseUrl = options.baseUrl || '';
 
     const request = async (signal?: AbortSignal) => {
-        const response = await httpPost(`${baseUrl}${url}`, body, { ...options, signal });
+        let response;
+        try {
+            response = await httpPost(`${baseUrl}${url}`, body, { ...options, signal });
+        } catch (e) {
+            // prevent dead cycles while using "deadline" option in scheduledAction
+            // catch fetch runtime errors like ECONNREFUSED or blocked by @trezor/request-manager
+            // and stop scheduledAction. those errors will not be resolved by retrying
+            if (options.deadline) {
+                return { error: e as Error };
+            }
+            throw e;
+        }
+
+        // throw unexpected network errors => retry scheduledAction
+        if (![200, 404, 500].includes(response.status)) {
+            if (response.status === 403 && options.identity) {
+                // NOTE: possibly blocked by cloudflare
+                // set random password to reset TOR circuit for this identity and then try again
+                const [user] = options.identity.split(':');
+                options.identity = `${user}:${Math.random()}`;
+            }
```

Comment:
:+1: Nice.
I hope cloudfare does not block many of our requests. It is weird to me since this service is intended to be used only from Tor network exit nodes. But I do not really know if cloudfare has some kind of config that will not block requests with the only reason that they come from Tor.

Maybe this is happening because the testnet coordinator is run by Trezor, but the one from mainnet is not going to have such blocking.